database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -44,13 +41,14 @@ public class SqlDatabaseService implements DatabaseService {
     static final String UNABLE_DELETE_ALL_RECORDS = "Unable to delete all records from table";
     static final String UNABLE_ACCESS_FIELD_VALUE = "Unable to access field value";
     static final String ID_PARAMETER_NAME = "id";
+    private Connection connection = null;
+    private Statement statement = null;
+    private ResultSet resultSet = null;
```

Comment:
Please don't use global variables for `connection`, `statement` and `resultSet`. That's why you get errors. You open connection in one method -> you close this connection in another method (for example, see `checkTableExists`) -> then you use already closed connection.